Pyrado/pyrado/algorithms/step_based/value_based.py

```diff
@@ -160,9 +166,20 @@ def memory(self) -> ReplayMemory:
     def step(self, snapshot_mode: str, meta_info: dict = None):
         if self._memory.isempty:
             # Warm-up phase
-            print_cbt_once("Collecting samples until replay memory if full.", "w")
+            print_cbt_once(f"Empty replay memory, collecting {self.num_init_memory_steps} samples.", "w")
             # Sample steps and store them in the replay memory
-            ros = self.sampler_init.sample()
+            if self._fill_with_init_sampler:
+                ros = self.sampler_init.sample()
+                self._fill_with_init_sampler = not self._use_trained_policy_for_refill
```

Comment:
wouldn't it be enough to check for `self._use_trained_policy_for_refill`?

Reply:
I think not, since we want to sample from the init sampler in the first iteration (at least I thought so). If this is the case you can interpret `self._fill_with_init_sampler` as `self._first_iteration_before_any_resets`.